tests/core/variant/test_array.h

```
@@ -160,19 +195,85 @@ TEST_CASE("[Array] remove_at()") {
 	CHECK(arr.size() == 0);
 }
 
-TEST_CASE("[Array] get()") {
-	Array arr = { 1 };
-	CHECK(int(arr.get(0)) == 1);
+TEST_CASE("[Array] get() and set()") {
+	Array arr = { 1, 2, 3 };
+	CHECK_EQ(int(arr.get(0)), 1);
+	CHECK_EQ(int(arr.get(1)), 2);
+	CHECK_EQ(int(arr.get(2)), 3);
+
+	arr.set(1, 5);
+	CHECK_EQ(int(arr.get(1)), 5);
```

I would also test "error" cases, such as `get()` and `set()` on a nonexistent index (surround it with `ERR_PRINT_OFF`/`ERR_PRINT_ON` to silence errors).

---

I tried adding these to the C++ tests, but it looks like calling `set` or `get` with an invalid index in C++ will just crash.  Binding for gdscript gets around this with some macros defined in variant_call.cpp, so it'll give an error message instead.

I added `array_bad_index` gdscript tests for this case for `get`/`set`/`operator[]`.  Not sure if this is expected, but the `operator[]` error handling is different from the others - `get()` and `set()` will print an error and continue running, while `operator[]` has a different error format and stops running.